src/main/java/uk/gov/companieshouse/authcodenotification/utils/Encrypter.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.companieshouse.authcodenotification.utils;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Base64;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+@Component
+public class Encrypter {
+
+    private static final String AES_CBC_PKCS_5_PADDING = "AES/CBC/PKCS5Padding";
+    private static final String AES = "AES";
+
+    @Value("${AUTH_CODE_ENCRYPT_KEY}")
+    private String aesKeyString;
+
+    public String encrypt(String plainText) throws
+            NoSuchPaddingException,
+            NoSuchAlgorithmException,
+            InvalidAlgorithmParameterException,
+            InvalidKeyException,
+            IllegalBlockSizeException,
+            BadPaddingException,
+            IOException {
+        byte[] key = aesKeyString.getBytes(StandardCharsets.UTF_8);
+        byte[] plaintextBytes = plainText.getBytes(StandardCharsets.UTF_8);
+
+        var cipher = Cipher.getInstance(AES_CBC_PKCS_5_PADDING);
+
+        var random = SecureRandom.getInstanceStrong();
+        var initialisationVector = new byte[Cipher.getInstance(AES_CBC_PKCS_5_PADDING).getBlockSize()];
+        random.nextBytes(initialisationVector);
+
+        cipher.init(Cipher.ENCRYPT_MODE, new SecretKeySpec(key, AES), new IvParameterSpec(initialisationVector));
+        byte[] encodedBytes = cipher.doFinal(plaintextBytes);
+
+        byte[] encodedSaltedBytes = concatByte(initialisationVector, encodedBytes);
+        return Base64.getEncoder().encodeToString(encodedSaltedBytes);
+    }
+
+    private byte[] concatByte(byte[] a, byte[] b) throws IOException {
```

Comment:
I would make the parameter names here a bit more descriptive.

Reply:
I'd agree, but can't think of a nice name - its a very generic functions that concats two byte arrays. Maybe byteArray1 and byteArray2, but not sure this adds much.

Reply:
Understand, but yes your suggestion is a slight improvement @pjames-ch .